### PR TITLE
transform any stable graph to graph of crates node and edges

### PR DESCRIPTION
**Pull request by chriamue**
This is a transformation of any stable graph to internal used stable graph of node and edges with data.
Use it as basis for a discussion.
I used this function in this example:
https://github.com/rusty-rails/transit-grid/blob/main/examples/visualize/src/main.rs

Another good feature would be to add a function reference that allows to customize a transformation of given weight to the data.

### CODE REVIEW

**Comment by blitzarx1** [comment]
Hey, thanks for this proposal. I ll take a look on the upcoming week.

**Comment by blitzarx1** [comment]
Nice. I have some comments to clarify why I choose mut ref to crates provided struct over any other options:

- First this project is just **widget** for egui ecosystem and should not be overcomplicated or store any huge state underhood of the egui runtime. It is not comprehensive standalone solution for graph data visualization and manipulation. 
- As egui widget the **GraphView** widget is initialized on every frame and it would be expensive to generate new state for it on every frame. That is why it is needed to be stored somewhere in the client app.
- If we already have a users graph with its data and widgets state with its own metadata (styling, coordinates, etc.) it is a good idea to combine them into one struct and let widget do all the needed things to its own metadata (not forbidden for user to manipulate with it either) and let user to handle his data in his own app.

Combining these points leads to a fact of having widget to accept a mut ref to widget provided struct which also holds user data. It should be generic over user data and have concrete metadata field which should be initialized on the user side.

**tldr** I liked the idea of helper methods for lighten the users burden to initialize this struct. I definitely had them on the list to be added. Thats why I am welcoming this pr and glad you used the project for your own.

To move on with this I will create a branch for the new release and reroute this pr there.
